### PR TITLE
	Removing require string inflections

### PR DESCRIPTION
**Pull request by lucasgomide**

### CODE REVIEW

File: lib/nifty_services/base_service.rb [hunk 1]
```diff
@@ -224,7 +226,7 @@ def process_error_message_for_key(message_key, options)
       elsif message_key.is_a?(Array) && message_key.first.is_a?(Hash)
         message = message_key
       else
-        message = I18n.t("#{i18n_errors_namespace}.#{message_key}", options)
+        message = translate("#{i18n_errors_namespace}.#{message_key}", options)
```
Comment (fidelisrafael) [inline lib/nifty_services/base_service.rb:229]: Good, now services can override this method to perform custom translations if needed. \o/

File: lib/nifty_services/support/string.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+unless String.method_defined?(:underscore)
+  class String
+   def underscore
```
Comment (fidelisrafael) [inline lib/nifty_services/support/string.rb:3]: @lucasgomide I think we can improve this method, for example, why are you using `freeze` in plain string objects? (I mean, this is usefull when you want to freeze a object who is assigned to a variable)
After this, ActiveSupport dependency will be removed \o/, thanks!

File: lib/nifty_services/support/string.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      return self unless self =~ /[A-Z-]|::/
+      word = self.to_s.gsub('::'.freeze, '/'.freeze)
+      word.gsub!(/(?:(?<=([A-Za-z\d]))|\b)(#{inflections.acronym_regex})(?=\b|[^a-z])/) { "#{$1 && '_'.freeze }#{$2.downcase}" }
```
Comment (fidelisrafael) [inline lib/nifty_services/support/string.rb]: I cant find a definition for this `inflections` method or variable, from where it comes?

Comment (lucasgomide) [inline lib/nifty_services/support/string.rb]: https://github.com/rails/rails/blob/f2489f493b794ee83a86e746b6240031acb8994e/activesupport/lib/active_support/inflector/methods.rb#L89

Comment (fidelisrafael) [inline lib/nifty_services/support/string.rb]: @lucasgomide Right, thanks, I get it! But keeping this we still depend on ActiveSupport, right?

File: lib/nifty_services/support/string.rb (hunk 2 above)
Comment (lucasgomide) [inline lib/nifty_services/support/string.rb:3]: This is usefull for Reducing object allocations too.

Ruby 2.2 already do that work for us

Comment (fidelisrafael) [inline lib/nifty_services/support/string.rb:3]: @lucasgomide So, this is ready for merge?

Comment (lucasgomide) [inline lib/nifty_services/support/string.rb:3]: yes bro

Comment (fidelisrafael) [inline lib/nifty_services/support/string.rb:3]: @lucasgomide thnks!

**Pull request merged**